cmds/ip/ip.go

```diff
@@ -200,11 +201,25 @@ func linkshow() {
 	}
 }
 
+func setHardwareAddress(iface netlink.Link) {
+	cursor++
+	hwAddr, err := net.ParseMAC(arg[cursor])
+	if err != nil {
+		log.Fatalf("%v cant parse mac addr %v: %v", iface, hwAddr, err)
```

Comment:
in the future, I would rather this return an error instead of log.Fatalfing here, but the whole thing is already written like this.

Reply:
To clarify your preference, would you like `setHardwareAddress` to return an error which is handled by `linkset`, or that `linkset` should also pass the error to be handled by `link`.  I think whichever you are requesting is an easy change (as I understand it), but I would like to clarify the point as how far the error should be propagated before being handled with Fatalf.

Reply:
My normal rule of thumb is the only main should log.Fatal, but that's some extra work for this command. I'm going to just merge this for now :)